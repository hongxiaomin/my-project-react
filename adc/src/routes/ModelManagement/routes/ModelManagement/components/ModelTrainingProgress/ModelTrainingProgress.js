import React from 'react'
import Chart from 're-echarts'
import { isEmpty } from 'lodash'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import classes from './ModelTrainingProgress.scss'

type Props = {
  lossRate: Array,
  accuracy: Array
};

const option = {
  tooltip: {
    trigger: 'axis',
  },
  legend: {
    data: ['Loss Rate', 'Accuracy'],
  },
  xAxis: [
    {
      type: 'category',
      data: [],
    },
  ],
  yAxis: [
    {
      type: 'value',
      name: 'Loss Rate',
      axisLabel: {
        formatter: '{value}',
      },
    },
    {
      type: 'value',
      name: 'Accuracy',
      axisLabel: {
        formatter: '{value}',
      },
    },
  ],
  series: [
    {
      name: 'Loss Rate',
      type: 'line',
      data: [],
    },
    {
      name: 'Accuracy',
      type: 'line',
      yAxisIndex: 1,
      data: [],
    },
  ],
}

export class ModelTrainingProgress extends React.Component {
  props: Props;

  render() {
    if (isEmpty(this.props.accuracy) || isEmpty(this.props.lossRate)) {
      return (
        <div className={classes['ModelTrainingProgress-container']}>
          <div style={{ color: '#337ab7', fontSize: '16px', fontWeight: 'bold' }}>
            <WrapperFormattedMessage id="labelingTool.dirTree.noData" />
          </div>
        </div>
      )
    }
    option.series[1].data = this.props.accuracy
    option.series[0].data = this.props.lossRate
    const tmp = []
    for (let i = 1; i <= this.props.lossRate.length; i += 1) {
      tmp.push(i)
    }
    option.xAxis[0].data = tmp
    return (
      <div className={classes['ModelTrainingProgress-container']}>
        <Chart option={option} style={{ height: '370px', overflow: 'none' }} />
      </div>
    )
  }
}

export default ModelTrainingProgress
