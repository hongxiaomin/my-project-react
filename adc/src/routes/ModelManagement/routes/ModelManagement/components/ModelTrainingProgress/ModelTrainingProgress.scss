.ModelTrainingProgress-container {
  // height: 400px;
  margin-top: 30px;
}
