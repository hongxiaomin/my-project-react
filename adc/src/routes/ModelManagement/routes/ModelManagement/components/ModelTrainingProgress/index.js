import ModelTrainingProgress from './ModelTrainingProgress'

export default ModelTrainingProgress
