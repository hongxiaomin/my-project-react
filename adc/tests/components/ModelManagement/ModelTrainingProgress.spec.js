import React from 'react'
import { shallow } from 'enzyme'
import { ModelTrainingProgress } from 'routes/ModelManagement/routes/ModelManagement/components/ModelTrainingProgress/ModelTrainingProgress'

describe('(Component) ModelTrainingProgress', () => {
  it('render normally', () => {
    const lossRate = [0.1, 0.2, 0.3]
    const accuracy = [0.5, 0.1, 0.8]
    const wrapper = shallow(
      <ModelTrainingProgress
        lossRate={lossRate}
        accuracy={accuracy}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
