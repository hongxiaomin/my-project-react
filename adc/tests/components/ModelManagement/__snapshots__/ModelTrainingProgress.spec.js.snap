// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ModelTrainingProgress render normally 1`] = `
<div
  className="ModelTrainingProgress-container"
>
  <ECharts
    config={Object {}}
    id="chart"
    lazyUpdate={false}
    notMerge={false}
    option={
      Object {
        "legend": Object {
          "data": Array [
            "Loss Rate",
            "Accuracy",
          ],
        },
        "series": Array [
          Object {
            "data": Array [
              0.1,
              0.2,
              0.3,
            ],
            "name": "Loss Rate",
            "type": "line",
          },
          Object {
            "data": Array [
              0.5,
              0.1,
              0.8,
            ],
            "name": "Accuracy",
            "type": "line",
            "yAxisIndex": 1,
          },
        ],
        "tooltip": Object {
          "trigger": "axis",
        },
        "xAxis": Array [
          Object {
            "data": Array [
              1,
              2,
              3,
            ],
            "type": "category",
          },
        ],
        "yAxis": Array [
          Object {
            "axisLabel": Object {
              "formatter": "{value}",
            },
            "name": "Loss Rate",
            "type": "value",
          },
          Object {
            "axisLabel": Object {
              "formatter": "{value}",
            },
            "name": "Accuracy",
            "type": "value",
          },
        ],
      }
    }
    style={
      Object {
        "height": "370px",
        "overflow": "none",
      }
    }
  />
</div>
`;
